File: easy/2490. Circular Sentence/2490. Circular Sentence.ts
function isCircularSentence(sentence: string): boolean {
    const senArr = sentence.split(" ")

    if (senArr.length === 1) {
        const curr = senArr[0]
        return curr[0] === curr.slice(-1)
    }

    for (let i = 0; i < senArr.length - 1; i++) {
        const curr = senArr[i]
        const next = senArr[i + 1]
        if (curr.slice(-1) !== next[0]) {
            return false
        }
    }

    const last = senArr.slice(-1)[0]
    return senArr[0][0] === last.slice(-1)
};

console.log(isCircularSentence("leetcode exercises sound delightful"))
console.log(isCircularSentence("eetcode"))
console.log(isCircularSentence("Leetcode is cool"))

File: easy/2490. Circular Sentence/2490. Circular Sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        sentence_arr = sentence.split(" ")

        if len(sentence) == 1:
            word = sentence_arr[0]
            return word[0] == word[-1]

        for index in range(0, len(sentence_arr) - 1):
            current = sentence_arr[index]
            next = sentence_arr[index + 1]

            if current[-1] != next[0]:
                return False

        return sentence_arr[0][0] == sentence_arr[-1][-1]


solution = Solution()
print(solution.isCircularSentence("leetcode exercises sound delightful"))  # True
print(solution.isCircularSentence("eetcode"))  # True
print(solution.isCircularSentence("Leetcode is cool"))  # False
